### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by Mctalian**
See commit description for changes. :beers:

### CODE REVIEW

**Comment by Mctalian** [comment]
@alanlgirard and @bcbrennecke and Andrew watched over my shoulder as I implemented and tested these chats changes.

@YashdalfTheGrey, @monotkate one of you take a sober look to make sure we're good here.

**Comment by Mctalian** [comment]
Damnit... @YashdalfTheGray

**Comment by Mctalian** [comment]
Also disregard the word "chats" a few comments above. Inadvertent word.

File: src/ideas/addIdeaView/addIdeaView.spec.js [hunk 1]
```diff
@@ -53,12 +53,12 @@ describe('AddIdeaViewCtrl', function() {
     it('should use the user\'s name as the author', function() {
         var idea = {
             title: 'Test Title',
-            author: 'Test',
```
Comment (Mctalian) [inline src/ideas/addIdeaView/addIdeaView.spec.js]: I made a mistake here. This should remain the same.

File: src/ideas/addIdeaView/addIdeaView.spec.js [hunk 2]
```diff
@@ -53,12 +53,12 @@ describe('AddIdeaViewCtrl', function() {
...
+            authorId: 3,
             description: 'This is a test idea.'
         };
         scope.addNewIdea(idea);
 
-        expect(idea.author).not.toBe("Test");
-        expect(idea.author).toBe("Darth Vader");
```
Comment (Mctalian) [inline src/ideas/addIdeaView/addIdeaView.spec.js]: This line should be .toBe(1)

Or whatever the Darth Vader _id  is in the login mock.

The line above it should be undeleted and the two new lines should be disregarded.

**Comment by monotkate** [comment]
Not sure it's necessary to route home if they aren't logged in for a new idea.  Maybe just through up an $mdToast message instead telling them they aren't logged in?  Should be able to login from any page.

I just tested this, you can log in from any page, but it will route you back to home after you're finished.  I'll fix this bug when I fix the lack of enter on login (I already know the fix for that)

**Comment by Mctalian** [comment]
This PR fixes that as well

**Comment by Mctalian** [comment]
"That" being the route back home on success. Not the enter button.

**Comment by monotkate** [comment]
Alright, cool.  I stand by notifying that they aren't logged in and staying on the same page instead of routing them back to home though for a 'need to be logged in' action

**Comment by Mctalian** [comment]
It's only if someone tries to go directly to the URL for adding a new idea
will it reroute to home. I do like the toast idea so maybe if someone does
try to access that url, it routes and toasts.

The "add new idea" button doesn't show up if they aren't logged in, so I
don't want to allow them to access it at all without being logged in.

**Comment by monotkate** [comment]
Agreed.  That makes sense for the redirect then, still digging my way through what all of the code does.  The rest of it looks good to me, Approved

**Comment by Mctalian** [comment]
Created Issue #138 to cover what we discussed here.

**Pull request merged**